frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.tsx

```diff
@@ -296,28 +250,50 @@ export function FieldValuesWidgetInner({
     }
   };
 
-  const fetchParameterValues = async (query?: string) => {
-    return fetchParameterValuesProp({
-      parameter,
-      query,
-    });
+  const _fetchParameterValues = async (query?: string) => {
+    if (!parameter) {
+      return { has_more_values: false, values: [] };
+    }
+
+    return dispatch(
+      fetchParameterValues({
+        parameter,
+        query,
+      }),
+    );
   };
 
-  const fetchCardParameterValues = async (query?: string) => {
-    return fetchCardParameterValuesProp({
-      cardId: question?.id(),
-      parameter,
-      query,
-    });
+  const _fetchCardParameterValues = async (query?: string) => {
+    const cardId = question?.id();
+
+    if (!cardId || !parameter) {
```

Comment:
`cardId` and `dashboardId` (below) can be numbers, so ideally we should handle the `0` as well.